From 0bae890b901d6382d31d9b72bc0deb7e633b3a0a Mon Sep 17 00:00:00 2001
From: Thiago Macieira <thiago.macieira@intel.com>
Date: Fri, 27 Nov 2020 08:22:46 -0800
Subject: [PATCH 107/171] qstring.h: fix warnings about shortening qsizetype to
 int

QStringView already uses qsizetype in Qt 5, which is 64-bit wide on 64-
bit platforms. But QString and its routines still use int.

Doesn't apply to Qt 6 since there QString has been updated to qsizetype.

Fixes: QTBUG-88952
Change-Id: I00b01c01a66748508ea1fffd164b69eee2079f96
Reviewed-by: Kai Koehne <kai.koehne@qt.io>
---
 src/corelib/text/qstring.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/corelib/text/qstring.h b/src/corelib/text/qstring.h
index fe97279aaf..8b71c014e0 100644
--- a/src/corelib/text/qstring.h
+++ b/src/corelib/text/qstring.h
@@ -2028,13 +2028,13 @@ inline int QStringRef::localeAwareCompare(const QString &s) const
 inline int QStringRef::localeAwareCompare(const QStringRef &s) const
 { return QString::localeAwareCompare_helper(constData(), length(), s.constData(), s.length()); }
 inline int QStringRef::localeAwareCompare(QStringView s) const
-{ return QString::localeAwareCompare_helper(constData(), length(), s.data(), s.size()); }
+{ return QString::localeAwareCompare_helper(constData(), length(), s.data(), int(s.size())); }
 inline int QStringRef::localeAwareCompare(const QStringRef &s1, const QString &s2)
 { return QString::localeAwareCompare_helper(s1.constData(), s1.length(), s2.constData(), s2.length()); }
 inline int QStringRef::localeAwareCompare(const QStringRef &s1, const QStringRef &s2)
 { return QString::localeAwareCompare_helper(s1.constData(), s1.length(), s2.constData(), s2.length()); }
 inline int QStringRef::localeAwareCompare(QStringView s1, QStringView s2)
-{ return QString::localeAwareCompare_helper(s1.data(), s1.size(), s2.data(), s2.size()); }
+{ return QString::localeAwareCompare_helper(s1.data(), int(s1.size()), s2.data(), int(s2.size())); }
 
 #if QT_STRINGVIEW_LEVEL < 2
 inline bool QStringRef::contains(const QString &s, Qt::CaseSensitivity cs) const
-- 
2.31.1

